packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -1019,6 +1020,9 @@ export class UserPool extends UserPoolBase {
 }
 
 function undefinedIfNoKeys(struct: object): object | undefined {
-  const allUndefined = Object.values(struct).reduce((acc, v) => acc && (v === undefined), true);
+  const allUndefined = Object.values(struct).every(val => val === undefined);
   return allUndefined ? undefined : struct;
 }
+function encodePunycodeIfDefined(input: string | undefined): string | undefined {
```

Comment:
Method name can be a bit terser. `IfDefined` is not necessary. Something like this maybe?

```suggestion
function encodePuny(input: string | undefined): string | undefined {
```